object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -308,22 +312,48 @@ private boolean isBuilderBuildMethod(
     }
   }
 
+  /**
+   * Is e a setter for an AutoValue builder?
+   *
+   * @param member member of builder or one of its supertypes
+   * @param builderElement element for the AutoValue builder
+   * @return {@code true} if e is a setter for the builder, {@code false} otherwise
+   */
+  private boolean isAutoValueBuilderSetter(Element member, Element builderElement) {
+    if (!member.getKind().equals(ElementKind.METHOD)) {
+      return false;
+    }
+    Set<Modifier> modifiers = member.getModifiers();
+    // should be abstract and not static
+    if (modifiers.contains(Modifier.STATIC) || !modifiers.contains(Modifier.ABSTRACT)) {
+      return false;
+    }
+    TypeMirror retType = ((ExecutableElement) member).getReturnType();
+    if (retType.getKind().equals(TypeKind.TYPEVAR)) {
+      // instantiate the type variable for the Builder class
+      retType =
+          AnnotatedTypes.asMemberOf(
+                  getContext().getTypeUtils(),
+                  ObjectConstructionAnnotatedTypeFactory.this,
+                  getAnnotatedType(builderElement),
+                  (ExecutableElement) member)
+              .getReturnType()
+              .getUnderlyingType();
+    }
+    // either the return type should be the builder itself, or it should be a Guava immutable type
+    return isGuavaImmutableType(retType)
+        || builderElement.equals(TypesUtils.getTypeElement(retType));
+  }
   /**
    * @param builderElement Element for a Lombok or AutoValue builder
    * @return names of all methods whose return type is the builder itself or returns a Guava
    *     Immutable type
    */
   private Set<String> getAllBuilderSetterMethodNames(Element builderElement) {
```

Comment:
Can you also change the name and documentation of this method to indicate that it only works for AutoValue now?